crud/program/src/instruction.rs

```diff
@@ -0,0 +1,135 @@
+//! Program instructions
+
+use crate::{state::Data, *};
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    pubkey::Pubkey,
+    sysvar,
+};
+
+/// Instructions supported by the Feature Proposal program
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, PartialEq)]
+pub enum CrudInstruction {
+    /// Create a new document
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    /// 0. `[writeable, signer]` Data account, must be uninitialized
+    /// 1. `[]` Document owner
+    /// 2. `[]` Rent sysvar, to check for rent exemption
+    Create {
```

Comment:
We typically use `Initialize` here instead of `Create`, since the account was created by the system program.
A couple comments:
1. The data account should not be the signer, as this means this program won't be compatible with `SystemInstruction::CreateAccountWithSeed`.
2. Provide an authority address to `Initialize` for the account owner.  This doesn't need be a signer

So like other programs, `Initialize` is now a fully unsigned instruction and must be issued in the same transaction as the account creation.

I feel like `data: Data` could potentially be removed for simplicity,  the user just immediately does an `Update` after `Initialize` (potentially within the same transaction for atomicity)

Reply:
That all makes sense.  Regarding point 2, how would that be different from the `1. Document owner` passed in?

Reply:
Done in e9b8f98 -- let me know if something else needs to be done on the authority